### PR TITLE
 Refactor tensor initializer

### PR DESCRIPTION
**Pull request by wyg1997**
这个 PR 对 Initializer 部分做了比较大的重构：

- 对齐 xavier_uniform、xavier_normal、kaiming_uniform_kaiming_normal、trunc_normal 公式以及文档重构
- oneflow.xx 顶级模块下删除所有 initializer 接口（这部分返回的是 oneflow 内部的 InitializerConf 对象，对用户没任何意义）
- flow.nn.init 模块下实现相关初始化具体逻辑，Tensor 下的初始化接口调用 flow.nn.init 中的接口（之前的代码正相反，Tensor 中实现逻辑，flow.nn.init 下调用 Tensor 的代码，不符合接口设计规范）
- 重新整理 initializer_conf.proto 中的配置项，删除无用/过时的项
- 把 initializer_util 和 initializer_register 分离，方便后续新的初始化代码添加/注册

fix oneflow-inc/onecloud#133

### CODE REVIEW

File: python/oneflow/ops/initializer_util.py [hunk 1]
```diff
@@ -1184,17 +933,17 @@ def VarianceScalingInitializerImpl(
...
-    scale = initializer_conf.scale / GenInitialFan(initializer_conf, var_blob_shape)
+    std = initializer_conf.scale / math.sqrt(GenInitialFan(initializer_conf, var_blob_shape))
     distribution = initializer_conf.distribution
     rng = np.random.default_rng(random_seed)
     if distribution == initializer_conf_util.kTruncatedNormal:
-        stddev = math.sqrt(scale) / 0.8796256610342398
-        return lambda length: RngTruncatedNormal(0, stddev, length, rng)
+        raise NotImplementedError
+        #  stddev = math.sqrt(scale) / 0.8796256610342398
+        #  return lambda length: RngTruncatedNormal(0, stddev, length, rng)
```
[the remark this answers is not in the document]
Comment (chengtbf) [reply python/oneflow/ops/initializer_util.py:942]: 我也忘记了，应该是当时 AlexNet loss 精度对齐的时候用的？

[the remark this answers is not in the document]
Comment (leaves-zwx) [reply python/oneflow/ops/initializer_util.py:942]: https://github.com/Oneflow-Inc/oneflow/pull/3540/files#diff-9d80350c62c92b84c6f19b3d6d0382ba96db5e2c7d7037be6682b5d1bf7629d9R1194

https://github.com/Oneflow-Inc/oneflow/pull/3540 我找到是 @daquexian 这个 pr 修改的，@chengbf 和 @liujuncheng review 的，看看他们还有没有什么印象

File: python/oneflow/framework/tensor.py [hunk 2]
```diff
@@ -237,50 +236,27 @@ def _uniform(self, a=0, b=1):
...
-    initializer_conf = flow.kaiming_initializer(
-        shape=self.shape,
-        distribution="random_uniform",
-        mode=mode,
-        nonlinearity=nonlinearity,
-        negative_slope=a,
-        data_format=data_format,
-    )
-    return _init_by_initializer_conf(self, initializer_conf)
+    return flow.nn.init.kaiming_normal_(self, a=a, mode=mode, nonlinearity=nonlinearity, data_format=data_format)
```
Comment (MARD1NO) [inline python/oneflow/framework/tensor.py:245]: 写反了吧，这个应该调用uniform？

Comment (wyg1997) [reply python/oneflow/framework/tensor.py]: 专门检察了一下还是漏了一个 🤣 ，已修正

**Review by MARD1NO — APPROVED** [review]

**Review by BBuf — APPROVED** [review]

**Comment by BBuf** [comment]
可以再注意一下文档渲染是否正常。

**Comment by wyg1997** [comment]
> 可以再注意一下文档渲染是否正常。

init 那部分的文档没有变，是李响刚入职时做的。我改的是内部才显示的文档，直接复制到相关网站上可以看公式，不影响我们内部看。

**Comment by wyg1997** [comment]
这次拿到了一个栈信息，但是不像是本 PR 引发的：

<details>
<summary>点击展开</summary>
<pre>
[wg=1][device=0,1,2,3] python/oneflow/test/modules/test_global_einsum_tensor_contraction.py F20220714 08:14:19.870118 34735 eager_nccl_kernels.cpp:364] 
[wg=1][device=0,1,2,3]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/user/kernels/eager_nccl_kernels.cpp", line 364, in Compute
[wg=1][device=0,1,2,3]     Recv<DeviceType::kCPU>( reinterpret_cast<void*>(reinterpret_cast ... elem_per_chunk, out->data_type(), src, ctx->stream())
[wg=1][device=0,1,2,3]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/user/kernels/communicate_util.cpp", line 58, in Recv
[wg=1][device=0,1,2,3]     ccl::Recv<device_type>(out, elem_cnt, dtype, src, stream)
[wg=1][device=0,1,2,3]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/ccl/ccl.cpp", line 553, in Recv<(oneflow::DeviceType)1>
[wg=1][device=0,1,2,3]     transport_ctx.WaitDone()
[wg=1][device=0,1,2,3]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/api/python/gil_foreign_lock_helper.cpp", line 31, in WithScopedRelease
[wg=1][device=0,1,2,3]     Callback()
[wg=1][device=0,1,2,3]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/common/blocking_counter.cpp", line 42, in operator()
[wg=1][device=0,1,2,3]     Check failed: cond_.wait_for(lck, seconds, [this]() { return cnt_val_ == 0; })
[wg=1][device=0,1,2,3] Error Type: oneflow.ErrorProto.timeout_error
[wg=1][device=0,1,2,3] *** Check failure stack trace: ***
[wg=1][device=0,1,2,3]     @     0x7f392d7663fa  google::LogMessage::Fail()
[wg=1][device=0,1,2,3]     @     0x7f392d7666e2  google::LogMessage::SendToLog()
[wg=1][device=0,1,2,3]     @     0x7f392d765f67  google::LogMessage::Flush()
[wg=1][device=0,1,2,3]     @     0x7f392d768ad9  google::LogMessageFatal::~LogMessageFatal()
[wg=1][device=0,1,2,3]     @     0x7f3927271727  oneflow::EagerCclS2SKernel<>::Compute()
[wg=1][device=0,1,2,3]     @     0x7f3928171d71  oneflow::one::StatefulOpKernel::Compute()
[wg=1][device=0,1,2,3]     @     0x7f3923c015a1  oneflow::vm::OpCallInstructionType::Compute()
[wg=1][device=0,1,2,3]     @     0x7f3926d37cd1  oneflow::vm::EventRecordedEpStreamType::Run()
[wg=1][device=0,1,2,3]     @     0x7f3926d3c5df  oneflow::vm::ThreadCtx::TryReceiveAndRun()
[wg=1][device=0,1,2,3]     @     0x7f3926d3e1d0  oneflow::(anonymous namespace)::WorkerLoop()
[wg=1][device=0,1,2,3] F20220714 08:14:20.382244 34736 eager_nccl_kernels.cpp:353] 
[wg=1][device=0,1,2,3]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/user/kernels/eager_nccl_kernels.cpp", line 353, in Compute
[wg=1][device=0,1,2,3]     Send<DeviceType::kCPU>( reinterpret_cast<const void ... elem_per_chunk, in->data_type(), dst, ctx->stream())
[wg=1][device=0,1,2,3]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/user/kernels/communicate_util.cpp", line 43, in Send
[wg=1][device=0,1,2,3]     ccl::Send<device_type>(in, elem_cnt, dtype, dst, stream)
[wg=1][device=0,1,2,3]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/ccl/ccl.cpp", line 514, in Send<(oneflow::DeviceType)1>
[wg=1][device=0,1,2,3]     transport_ctx.WaitDone()
[wg=1][device=0,1,2,3]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/api/python/gil_foreign_lock_helper.cpp", line 31, in WithScopedRelease
[wg=1][device=0,1,2,3]     Callback()
[wg=1][device=0,1,2,3]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/common/blocking_counter.cpp", line 42, in operator()
[wg=1][device=0,1,2,3]     Check failed: cond_.wait_for(lck, seconds, [this]() { return cnt_val_ == 0; })
[wg=1][device=0,1,2,3] Error Type: oneflow.ErrorProto.timeout_error
[wg=1][device=0,1,2,3]     @     0x7f3926d3e32d  _ZNSt6thread11_State_implINS_8_InvokerISt5tupleIJPFvPN7oneflow2vm9ThreadCtxERKSt8functionIFvS6_EEES6_ZNS3_14VirtualMachine15CreateThreadCtxENS3_6SymbolINS3_6DeviceEEENS3_10StreamRoleEEUlS6_E0_EEEEE6_M_runEv
[wg=1][device=0,1,2,3] *** Check failure stack trace: ***
[wg=1][device=0,1,2,3]     @     0x7f392d77ae3f  execute_native_thread_routine
[wg=1][device=0,1,2,3]     @     0x7f39907b76db  start_thread
[wg=1][device=0,1,2,3]     @     0x7f2d316493fa  google::LogMessage::Fail()
[wg=1][device=0,1,2,3]     @     0x7f39904e061f  clone
[wg=1][device=0,1,2,3] Fatal Python error: Aborted
[wg=1][device=0,1,2,3] 
[wg=1][device=0,1,2,3] Thread 0x00007f37b0ffd700 (most recent call first):
[wg=1][device=0,1,2,3] 
[wg=1][device=0,1,2,3] Thread 0x00007f37b1fff700 (most recent call first):
[wg=1][device=0,1,2,3] 
[wg=1][device=0,1,2,3] Thread 0x00007f37b17fe700 (most recent call first):
[wg=1][device=0,1,2,3] 
[wg=1][device=0,1,2,3] Thread 0x00007f37d8aef700 (most recent call first):
[wg=1][device=0,1,2,3] 
[wg=1][device=0,1,2,3] Thread 0x00007f399138f0c0 (most recent call first):
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/posixpath.py", line 361 in normpath
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/posixpath.py", line 385 in abspath
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/posixpath.py", line 396 in realpath
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/inspect.py", line 742 in getmodule
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/inspect.py", line 780 in findsource
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/_code/source.py", line 121 in findsource
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/_code/code.py", line 106 in fullsource
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/_code/code.py", line 250 in getsource
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/_code/code.py", line 718 in _getentrysource
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/_code/code.py", line 810 in repr_traceback_entry
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/_code/code.py", line 867 in repr_traceback
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/_code/code.py", line 926 in repr_excinfo
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/_code/code.py", line 666 in getrepr
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/nodes.py", line 481 in _repr_failure_py
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/python.py", line 1795 in repr_failure
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/reports.py", line 345 in from_item_and_call
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/runner.py", line 365 in pytest_runtest_makereport
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/pluggy/_callers.py", line 39 in _multicall
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/pluggy/_manager.py", line 80 in _hookexec
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/pluggy/_hooks.py", line 265 in __call__
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/runner.py", line 221 in call_and_report
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/runner.py", line 130 in runtestprotocol
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/runner.py", line 111 in pytest_runtest_protocol
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/pluggy/_callers.py", line 39 in _multicall
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/pluggy/_manager.py", line 80 in _hookexec
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/pluggy/_hooks.py", line 265 in __call__
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/main.py", line 347 in pytest_runtestloop
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/pluggy/_callers.py", line 39 in _multicall
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/pluggy/_manager.py", line 80 in _hookexec
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/pluggy/_hooks.py", line 265 in __call__
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/main.py", line 322 in _main
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/main.py", line 268 in wrap_session
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/main.py", line 315 in pytest_cmdline_main
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/pluggy/_callers.py", line 39 in _multicall
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/pluggy/_manager.py", line 80 in _hookexec
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/pluggy/_hooks.py", line 265 in __call__
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/config/__init__.py", line 165 in main
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/config/__init__.py", line 187 in console_main
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/pytest/__main__.py", line 5 in <module>
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/runpy.py", line 85 in _run_code
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/runpy.py", line 193 in _run_module_as_main
[wg=1][device=0,1,2,3]     @     0x7f2d316496e2  google::LogMessage::SendToLog()
[wg=1][device=0,1,2,3]     @     0x7f2d31648f67  google::LogMessage::Flush()
[wg=1][device=0,1,2,3]     @     0x7f2d3164bad9  google::LogMessageFatal::~LogMessageFatal()
[wg=1][device=0,1,2,3]     @     0x7f2d2b153ddd  oneflow::EagerCclS2SKernel<>::Compute()
[wg=1][device=0,1,2,3]     @     0x7f2d2c054d71  oneflow::one::StatefulOpKernel::Compute()
[wg=1][device=0,1,2,3]     @     0x7f2d27ae45a1  oneflow::vm::OpCallInstructionType::Compute()
[wg=1][device=0,1,2,3]     @     0x7f2d2ac1acd1  oneflow::vm::EventRecordedEpStreamType::Run()
[wg=1][device=0,1,2,3]     @     0x7f2d2ac1f5df  oneflow::vm::ThreadCtx::TryReceiveAndRun()
[wg=1][device=0,1,2,3] F20220714 08:14:20.382261 34734 eager_nccl_kernels.cpp:353] 
[wg=1][device=0,1,2,3]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/user/kernels/eager_nccl_kernels.cpp", line 353, in Compute
[wg=1][device=0,1,2,3]     Send<DeviceType::kCPU>( reinterpret_cast<const void ... elem_per_chunk, in->data_type(), dst, ctx->stream())
[wg=1][device=0,1,2,3]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/user/kernels/communicate_util.cpp", line 43, in Send
[wg=1][device=0,1,2,3]     ccl::Send<device_type>(in, elem_cnt, dtype, dst, stream)
[wg=1][device=0,1,2,3]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/ccl/ccl.cpp", line 514, in Send<(oneflow::DeviceType)1>
[wg=1][device=0,1,2,3]     transport_ctx.WaitDone()
[wg=1][device=0,1,2,3]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/api/python/gil_foreign_lock_helper.cpp", line 31, in WithScopedRelease
[wg=1][device=0,1,2,3]     Callback()
[wg=1][device=0,1,2,3]   File "/home/ci-user/runners/release/_work/oneflow/oneflow/oneflow/core/common/blocking_counter.cpp", line 42, in operator()
[wg=1][device=0,1,2,3]     Check failed: cond_.wait_for(lck, seconds, [this]() { return cnt_val_ == 0; })
[wg=1][device=0,1,2,3] Error Type: oneflow.ErrorProto.timeout_error
[wg=1][device=0,1,2,3]     @     0x7f2d2ac211d0  oneflow::(anonymous namespace)::WorkerLoop()
[wg=1][device=0,1,2,3] *** Check failure stack trace: ***
[wg=1][device=0,1,2,3]     @     0x7fd6535333fa  google::LogMessage::Fail()
[wg=1][device=0,1,2,3]     @     0x7fd6535336e2  google::LogMessage::SendToLog()
[wg=1][device=0,1,2,3]     @     0x7fd653532f67  google::LogMessage::Flush()
[wg=1][device=0,1,2,3]     @     0x7f2d2ac2132d  _ZNSt6thread11_State_implINS_8_InvokerISt5tupleIJPFvPN7oneflow2vm9ThreadCtxERKSt8functionIFvS6_EEES6_ZNS3_14VirtualMachine15CreateThreadCtxENS3_6SymbolINS3_6DeviceEEENS3_10StreamRoleEEUlS6_E0_EEEEE6_M_runEv
[wg=1][device=0,1,2,3]     @     0x7f2d3165de3f  execute_native_thread_routine
[wg=1][device=0,1,2,3]     @     0x7fd653535ad9  google::LogMessageFatal::~LogMessageFatal()
[wg=1][device=0,1,2,3]     @     0x7f2d9469a6db  start_thread
[wg=1][device=0,1,2,3]     @     0x7fd64d03dddd  oneflow::EagerCclS2SKernel<>::Compute()
[wg=1][device=0,1,2,3]     @     0x7f2d943c361f  clone
[wg=1][device=0,1,2,3] Fatal Python error: Aborted
[wg=1][device=0,1,2,3] 
[wg=1][device=0,1,2,3] Thread 0x00007f2badff[2700](https://github.com/Oneflow-Inc/oneflow/runs/7335332648?check_suite_focus=true#step:40:2701) (most recent call first):
[wg=1][device=0,1,2,3] 
[wg=1][device=0,1,2,3] Thread 0x00007f2bae7f3700 (most recent call first):
[wg=1][device=0,1,2,3] 
[wg=1][device=0,1,2,3] Thread 0x00007f2baeff4700 (most recent call first):
[wg=1][device=0,1,2,3] 
[wg=1][device=0,1,2,3] Thread 0x00007f2bdc9dc700 (most recent call first):
[wg=1][device=0,1,2,3] 
[wg=1][device=0,1,2,3] Thread 0x00007f2d95[2720](https://github.com/Oneflow-Inc/oneflow/runs/7335332648?check_suite_focus=true#step:40:2721)c0 (most recent call first):
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/posixpath.py", line 415 in _joinrealpath
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/posixpath.py", line 395 in realpath
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/inspect.py", line 742 in getmodule
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/inspect.py", line 780 in findsource
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/_code/source.py", line 121 in findsource
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/_code/code.py", line 106 in fullsource
[wg=1][device=0,1,2,3]   File "/opt/conda/lib/python3.7/site-packages/_pytest/_code/code.py", line 250 in getsource
</pre>
</details>

**Pull request merged**